objectivec/cxx_utils.mm

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "cxx_utils.h"
+
+#import <vector>
+#import <optional>
+#import <string>
+
+#import "error_utils.h"
+
+#import "ort_value_internal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace utils {
+
+NSString* _Nullable toNSString(const std::string& str) {
+  return [NSString stringWithUTF8String:str.c_str()];
+}
+
+NSString* _Nullable toNullableNSString(const std::optional<std::string>& str) {
+  if (str.has_value()) {
+    return [NSString stringWithUTF8String:str->c_str()];
+  }
+  return nil;
+}
+
+std::string toStdString(NSString* str) {
+  return std::string([str UTF8String]);
+}
+
+std::optional<std::string> toStdOptionalString(NSString* _Nullable str) {
+  if (str) {
+    return std::optional<std::string>([str UTF8String]);
+  }
+  return std::nullopt;
+}
+
+std::vector<std::string> toStdStringVector(NSArray<NSString*>* strs) {
+  std::vector<std::string> result;
+  result.reserve(strs.count);
+  for (NSString* str in strs) {
+    result.push_back([str UTF8String]);
+  }
+  return result;
+}
+
+NSArray<NSString*>* toNSStringNSArray(const std::vector<std::string>& strs) {
+  NSMutableArray<NSString*>* result = [NSMutableArray arrayWithCapacity:strs.size()];
+  for (const std::string& str : strs) {
+    NSString* nsStr = [NSString stringWithUTF8String:str.c_str()];
+    if (nsStr) {
+      [result addObject:nsStr];
+    } else {
+      ORT_CXX_API_THROW("Failed to convert std::string to NSString", ORT_INVALID_ARGUMENT);
+    }
```

Comment:
maybe we can do a similar check in toNSString so it doesn't need to have a nullable return type?